modules/stochastic_tools/test/tests/surrogates/libtorch_nn/train.i

```diff
@@ -33,6 +33,7 @@
     read_from_file = false
     print_epoch_loss = 10
     activation_function = 'relu relu'
+    max_processes = 1
```

Comment:
Does it not run with multiple procs and/or produce different results? Or will it run the same training process on multiple procs and it becomes a waste of resources?

Reply:
Yes, with multiple processors, it will generate different results considering that the gradient is computed on subsets on each processor and then the gradient is averaged. So in the update part, the gradients will be different so the results may be different too. Also if we have a large Moose problem which needs, let's say, 500 processors and we only get 200 samples out of it, we would end up with more samples than processes which would error out in the training of the neural net. This parameter can limit the number of processors used for the neural net training.

Reply:
Cool, thanks for the explanation!